include/poggles/gltypes.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <concepts>
+
+#include <glad/glad.h>
+
+#include "poggles/poggles_export.hpp"
+
+template<typename T>
+concept GLuintValued = requires(T a) {
```

Comment:
We should format this in an easy to read manner and turn off clang-format for the concept.

Reply:
Please also format this one, and then we're good to merge.

Reply:
LGTM